### ISSUE TITLE
vit_base_patch16_siglip_256 amp training accuracy regression on H100 (2.13.0)

### ISSUE THREAD

**Issue by huydhn** [pytorch/pytorch#188912]
## Summary

`vit_base_patch16_siglip_256` (timm) **amp training** accuracy regressed to `fail_accuracy` in the **2.13.0** release candidate on H100. It **passes on 2.12.1**, **fails on 2.13.0**, and **passes again on current `main`**. All three inductor configs fail (`inductor_default`, `inductor_dynamic_shapes`, `inductor_cudagraphs`). It shows up on the TorchInductor perf dashboard (`inductor-perf-test-nightly-h100`, timm training) as `pass -> fail_accuracy`.

This is a numerical accuracy mismatch, not a crash — the compiled scalar gradient diverges from the fp64 reference:

```
E torch/_dynamo/utils.py] RMSE (res-fp64): 0.38343, (ref-fp64): 0.09728 and shape=torch.Size([]).
  res.dtype: torch.float32, multiplier: 3.000000, tol: 0.080000
fail_accuracy
```

## Events on `main`

| Date | Commit / PR | Description | Effect on siglip amp-train |
|---|---|---|---|
| 2026-04-15 | `58e3fb03f67` | 2.12 branch cut | — |
| ~2026-04 | v2.12.1 | release | pass |
| 2026-05-20 | `b2bc2b83bce` #182897 | Lower nested reductions in SIMD codegen | **-> fail** (root cause) |
| 2026-05-26 | `93cce594152` #183680 | addmm narrowing-cast unfuse for AMP | -> pass (incidentally masks #182897) |
| 2026-06-02 | `a8e63623df8` #185856 | gate that unfuse to XPU-only (removes it for CUDA) | **-> fail** (unmasks #182897) |
| 2026-06-10 | `636bc9106d0` | 2.13 branch cut (has #182897 + #183680 + #185856) | — |
| ~2026-06 | 2.13.0 | release | fail |
| 2026-06-24 | `f41aca09a57` #183661 | L2-aware two-pass variance heuristic | -> pass (landed after the 2.13 branch cut) |
| now | `main` (2.14-dev) | has #182897 + #185856 + #183661 | pass |

Root cause is **#182897** (nested-reduction SIMD codegen), which degraded the amp reduction behind siglip's scalar gradient. **#183680** briefly masked it; **#185856** re-exposed it for CUDA. 2.13.0 shipped with #182897 + #185856 but without the later change (#183661) that happens to make `main` pass, because #183661 landed ~2 weeks after the 2.13 branch cut.

## Reproducer (H100)

This only reproduces on a **cuDNN-enabled build on a full-SM H100** (official wheels; cuDNN `9.20.0.48`). It is gated on the cuDNN / `is_big_gpu` Triton codegen path — cuDNN-disabled builds and reduced-SM / MIG GPUs pass and do not reproduce it.

```bash
# torch==2.13.0 (download.pytorch.org/whl/test/cu130) on an H100
python benchmarks/dynamo/timm_models.py --accuracy --no-translation-validation \
  --training --amp --backend inductor --disable-cudagraphs --device cuda \
  --only vit_base_patch16_siglip_256
```

Expected: `fail_accuracy` with `RMSE (res-fp64): 0.38343`. The `--dynamic-shapes --dynamic-batch-only` and cudagraphs (drop `--disable-cudagraphs`) variants fail identically.

## Note on #183661

`main` currently passes because of #183661. However, per https://github.com/pytorch/pytorch/pull/183661#issuecomment-4869570415 it is causing regressions elsewhere and is likely to be reverted, so it is **not** ready to be a cherry-pick candidate for 2.13.0 yet.

cc @csarofeen @ptrblck @eqy @nWEIdia @mcarilli @leslie-fang-intel @jgong5 @chauhang @penguinwu @voznesenskym @EikanWang @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by huydhn** [comment]
We can resolve this issue. This looks fixed in main

```
python benchmarks/dynamo/timm_models.py --accuracy --no-translation-validation \
    --training --amp --backend inductor --disable-cudagraphs --device cuda \
    --only vit_base_patch16_siglip_256
/home/dev/venv/py3.12/lib/python3.12/site-packages/torch/jit/_script.py:1643: FutureWarning: `torch.jit.interface` is deprecated. Please use `torch.compile` instead.
  warnings.warn(
model.safetensors: 100%|████████████████████████████████████| 372M/372M [00:01<00:00, 296MB/s]
loading model: 0it [00:04, ?it/s]███████████████████████████| 372M/372M [00:01<00:00, 451MB/s]
cuda train vit_base_patch16_siglip_256
W0806 22:44:28.976000 1213 torch/_logging/_internal.py:1394] [5/0] Profiler record function <class 'torch.autograd.profiler.record_function'> will be ignored
pass
```
